controllers/handlerFactory.js

```diff
@@ -46,7 +48,17 @@ exports.getOne = (Model, popOptions) =>
   catchAsync(async (req, res, next) => {
     let query = Model.findById(req.params.id);
     if (popOptions)
-      query = query.populate(popOptions).populate('author').exec();
+      query = query
+        .populate({
+          path: popOptions,
+          model: 'resources',
+          populate: {
+            path: 'images',
```

Comment:
will his populate an array of images?

Reply:
yes, thanks  to references in mongo

Reply:
ok, I was confused by their way of documenting it here, I think the second section is continuing to explain the first section, so the empty array is in case of none existing authors.
https://mongoosejs.com/docs/populate.html#doc-not-found